### PR TITLE
Assignments Finalization

### PR DESCRIPTION
**Pull request by nathanielparke**
Adds in assignments finalization framework.

### CODE REVIEW

**Comment by nathanielparke** [comment]
screen shot 2015-11-14 at 3 40 07 pm

**Comment by nathanielparke** [comment]
screen shot 2015-11-14 at 3 40 15 pm

**Comment by nathanielparke** [comment]
screen shot 2015-11-14 at 3 40 22 pm

File: huxley/api/urls.py [hunk 1]
```diff
@@ -27,7 +27,14 @@
...
     url(r'^schools/(?P<pk>[0-9]+)/?$', views.school.SchoolDetail.as_view(), name='school_detail'),
     url(r'^schools/(?P<pk>[0-9]+)/assignments/?$', views.school.SchoolAssignments.as_view(), name='school_assignments'),
+    url(r'^schools/(?P<pk>[0-9]+)/assignments/finalize/?$', views.school.SchoolAssignmentsFinalize.as_view(), name='school_assignments_finalize'),
     url(r'^schools/(?P<pk>[0-9]+)/invoice/?$', views.school.SchoolInvoice.as_view(), name='school_invoice')
 )
 
+urlpatterns += patterns('',
+    url(r'^assignments/?$', views.assignment.AssignmentList.as_view(), name='assignment_list'),
+    url(r'^assignments/(?P<pk>[0-9]+)/?$', views.assignment.AssignmentDetail.as_view(), name='assignment_detail'),
+    url(r'^assignments/(?P<pk>[0-9]+)/delete/?$', views.assignment.AssignmentDelete.as_view(), name='assignment_delete'),
```
Comment (nathanielparke) [inline huxley/api/urls.py]: Probably should remove this and wait till delete feature is built.

File: huxley/api/views/assignment.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+class AssignmentDelete(generics.GenericAPIView):
+    authentication_classes = (SessionAuthentication,)
+    permission_classes = (IsSchoolAdvisorOrSuperuser,)
+
+    def post(self, request, *args, **kwargs):
+        assignment_id = kwargs.get('pk', None)
+        assignment = Assignment.objects.get(id=assignment_id)
+        assignment.delete()
+
+        return Response(status=status.HTTP_200_OK)
```
Comment (nathanielparke) [inline huxley/api/views/assignment.py]: Also remove this and move to new diff.

File: huxley/www/static/js/huxley/components/AdvisorAssignmentsView.js [hunk 3]
```diff
@@ -64,11 +72,24 @@ var AdvisorAssignmentsView = React.createClass({
                 <th>Committee</th>
                 <th>Country</th>
                 <th>Delegation Size</th>
+                <th>{this.state.finalized ?
+                  "" :
+                  "Relinquish Assignment"}
+                </th>
```
Comment (nathanielparke) [inline huxley/www/static/js/huxley/components/AdvisorAssignmentsView.js]: Move to new commit

File: huxley/www/static/js/huxley/components/AdvisorAssignmentsView.js [hunk 4]
```diff
@@ -83,9 +104,64 @@ var AdvisorAssignmentsView = React.createClass({
...
+        url: '/api/assignments/'+assignment.id+'/delete/',
+        success: this._handleSuccess,
+        error: this._handleError,
+      });
+      var user = this.props.user.getData();
+      AssignmentStore.getAssignments(user.school.id, function(assignments) {
+        this.setState({assignments: assignments});
+      }.bind(this));
+    }
+  },
```
Comment (nathanielparke) [inline huxley/www/static/js/huxley/components/AdvisorAssignmentsView.js]: Move to next commit

File: huxley/www/static/js/huxley/components/AdvisorAssignmentsView.js [hunk 5]
```diff
@@ -83,9 +104,64 @@ var AdvisorAssignmentsView = React.createClass({
...
+          </td>
         </tr>
-      );
-    });
+      )
+    }.bind(this));
+  },
+
+  _handleFinalize: function(event) {
+    var confirm = window.confirm("By pressing okay you are committing to the financial respoinsibility of each assingment. Are you sure you want to finalize assignments?");
```
Comment (nathanielparke) [inline huxley/www/static/js/huxley/components/AdvisorAssignmentsView.js]: 'respoinsibility' spelled wrong

File: huxley/settings/logging.py [hunk 6]
```diff
@@ -40,7 +40,7 @@
         'huxley.api': {
             'handlers': ['database'],
             'level': 'DEBUG',
-            'propagate': True, #Set to false to elimninate logs in the console
+            'propagate': False, #Set to false to elimninate logs in the console
```
Comment (kmeht) [inline huxley/settings/logging.py]: Is this line supposed to be here?

**Comment by kmeht** [comment]
Looks good to me, but it seems like for the finalize API, we're sending the `{assignments_finalized: true}` as JSON and the view is performing a partial update. Couldn't we just make a generic partial update API for the school and send whatever we want to it?

In fact, that's what'll happen now. We can send literally any school field to the finalize view and it will accept it, right?

File: huxley/settings/logging.py (hunk 6 above)
Comment (nathanielparke) [inline huxley/settings/logging.py]: It is just to keep the login suppressed when running the tests. We need to change the way that we are logging this data at some point.

**Pull request merged**